src/Microsoft.Extensions.CodeGenerators.Mvc/Controller/ControllerWithContextGenerator.cs

```diff
@@ -26,8 +40,87 @@ public abstract class ControllerWithContextGenerator : ControllerGeneratorBase
             EntityFrameworkService = entityFrameworkService;
         }
 
+        public override async Task Generate(CommandLineGeneratorModel controllerGeneratorModel)
+        {
+            Contract.Assert(!String.IsNullOrEmpty(controllerGeneratorModel.ModelClass));
+
+            string outputPath = ValidateAndGetOutputPath(controllerGeneratorModel);
+            var modelTypeAndContextModel = await ValidateModelAndGetMetadata(controllerGeneratorModel);
+
+            if (string.IsNullOrEmpty(controllerGeneratorModel.ControllerName))
+            {
+                //Todo: Pluralize model name
+                controllerGeneratorModel.ControllerName = modelTypeAndContextModel.ModelType.Name + Constants.ControllerSuffix;
+            }
+
+            var templateModel = new ControllerWithContextTemplateModel(modelTypeAndContextModel.ModelType, modelTypeAndContextModel.DbContextFullName)
+            {
+                ControllerName = controllerGeneratorModel.ControllerName,
+                AreaName = string.Empty, //ToDo
+                UseAsync = controllerGeneratorModel.UseAsync,
+                ControllerNamespace = GetControllerNamespace(),
+                ModelMetadata = modelTypeAndContextModel.ModelMetadata
+            };
+
+            await CodeGeneratorActionsService.AddFileFromTemplateAsync(outputPath, GetTemplateName(controllerGeneratorModel), TemplateFolders, templateModel);
+            Logger.LogMessage("Added Controller : " + outputPath.Substring(ApplicationEnvironment.ApplicationBasePath.Length));
+
+            await GenerateViewsIfRequired(controllerGeneratorModel, modelTypeAndContextModel, templateModel.ControllerRootName);
+        }
+
+        private async Task GenerateViewsIfRequired(CommandLineGeneratorModel controllerGeneratorModel,
+            ModelTypeAndContextModel modelTypeAndContextModel,
+            string controllerRootName)
+        {
+            if (!controllerGeneratorModel.IsRestController && !controllerGeneratorModel.NoViews)
+            {
+                var layoutDependencyInstaller = ActivatorUtilities.CreateInstance<MvcLayoutDependencyInstaller>(ServiceProvider);
+                await layoutDependencyInstaller.Execute();
+
+                foreach (var viewTemplate in _views)
+                {
+                    var viewName = viewTemplate == "List" ? "Index" : viewTemplate;
```

Comment:
no one else I using _views, why we have List there? Why not have Index? Any sort of convention?